barman/copy_controller.py

```diff
@@ -900,6 +902,16 @@ def _analyze_directory(self, item):
             self.temp_dir, "%s_exclude_and_protect.filter" % item.label
         )
         exclude_and_protect_filter = open(item.exclude_and_protect_file, "w+")
+
+        if not ref_has_content:
+            # If the destination directory is empty then include all
+            # directories and exclude all files. This stops the rsync
+            # command which runs during the _create_dir_and_purge function
+            # from copying the entire contents of the source directory and
+            # ensures it only creates the directories.
```

Comment:
I like the  explanation +1 for readability. 
Can't say about the behavior though, I haven't spent enough time to understand it.